Breadcrumbs.py

```diff
@@ -32,80 +38,169 @@ def get_row_indentation(points, view, tab_size, limit=1e20):
+    points = xrange(line_start, last_line_start)
+    last_line_start = line_start
+    current_indentation = get_row_indentation(points, view, tab_size, indentation)
+    if current_indentation < indentation and not is_white_row(view, points):
+      indentation = current_indentation
+      breadcrumbs.append(get_breadcrumb(view, points, my_regex, separator))
+
+    current_row -= 1
+
+  breadcrumbs.reverse()
+  lengths = [len(breadcrumb) for breadcrumb in breadcrumbs]
+  sorted_lengths = sorted(lengths)
+  previous_length = 0
+  number_of_characters_left  = total_breadcrumbs_length_limit - len(lengths) * len(separator)
+  for (number_of_shorter, current_length) in enumerate(sorted_lengths):
+    if previous_length < current_length:
+      previous_length = current_length
+      number_of_not_shorter = len(breadcrumbs) - number_of_shorter
+      if number_of_characters_left < current_length * number_of_not_shorter:
+        length_of_short_trim = number_of_characters_left // number_of_not_shorter
+        length_of_long_trim = length_of_short_trim + 1
+        number_of_long_trimmed = number_of_characters_left % number_of_not_shorter
+        number_of_short_trimmed = number_of_not_shorter - number_of_long_trimmed
+        for (index_of_breadcrumb, breadcrum_length) in enumerate(lengths):
+          if current_length <= breadcrum_length:
+            if 0 < number_of_short_trimmed:
+              length_of_trim = length_of_short_trim
+              number_of_short_trimmed -= 1
+            else:
+              length_of_trim = length_of_long_trim
+            breadcrumbs[index_of_breadcrumb] = breadcrumbs[index_of_breadcrumb][0:length_of_trim]
+        break
+    number_of_characters_left -= current_length
+
+  return ''.join(breadcrumbs+[''])
+
+
+stylesheet = '''
+  <style>
+    div.phantom-arrow {
+      border-top: 0.4rem solid transparent;
+      border-left: 0.5rem solid color(var(--bluish) blend(var(--background) 30%));
+      width: 0;
+      height: 0;
+    }
+    div.phantom {
+      padding: 0.4rem 0 0.4rem 0.7rem;
+      margin: 0 0 0.2rem;
+      border-radius: 0 0.2rem 0.2rem 0.2rem;
+      background-color: color(var(--bluish) blend(var(--background) 30%));
+    }
+    div.phantom span.message {
+      padding-right: 0.7rem;
+    }
+    div.phantom a {
+      text-decoration: inherit;
+    }
+    div.phantom a.close {
+      padding: 0.35rem 0.7rem 0.45rem 0.8rem;
+      position: relative;
+      bottom: 0.05rem;
+      border-radius: 0 0.2rem 0.2rem 0;
+      font-weight: bold;
+    }
+    html.dark div.phantom a.close {
+      background-color: #00000018;
+    }
+    html.light div.phantom a.close {
+      background-color: #ffffff18;
+    }
+  </style>
+'''
+
+
+template = '''
+  <body>
+    {stylesheet}
+    <div class="phantom-arrow"></div>
+    <div class="phantom">
+      <span class="message">
+        <strong>Breadcrumbs:</strong> {breadcrumbs}
+        <a class="close" href="close">''' + chr(0x00D7) + '''</a>
```

Comment:
I was wondering if we could arrange it in such a way, that clicking on a particular breadcrumb would make you jump to the line of that breadcrumb. But that would definitely be out of scope of this PR. 

As to "close on ESC". As you've said in a different place: it wouldn't be good to force key bindings on users. So I'd make it configurable. My personal preference would be to have the same keys combination to show and hide the phantoms, this way I could easily toggle between the two states without moving my fingers. So if I just want to show phantoms for a second I would just hit some combo, and then hit it again to close them. But if I were in a mood to just keep them open while moving around the code, I could also do so using the same combination. 
Ideally, this should all be configurable.